sylvia-derive/src/input.rs

```diff
@@ -25,12 +27,14 @@ pub struct TraitInput<'a> {
     item: &'a ItemTrait,
     generics: Vec<&'a GenericParam>,
     custom: Custom<'a>,
+    base_exec: Option<Type>,
```

Comment:
This allows for registering a base execution method that will be triggered before executing the trait message